kroxylicious-filters/kroxylicious-oauthbearer/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerFilter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.oauthbearer;
+
+import java.util.concurrent.CompletionStage;
+
+import javax.security.sasl.SaslServer;
+
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.message.SaslAuthenticateRequestData;
+import org.apache.kafka.common.message.SaslAuthenticateResponseData;
+
+import io.kroxylicious.proxy.filter.FilterContext;
+import io.kroxylicious.proxy.filter.RequestFilterResult;
+import io.kroxylicious.proxy.filter.SaslAuthenticateRequestFilter;
+
+import static org.apache.kafka.common.protocol.Errors.SASL_AUTHENTICATION_FAILED;
+
+public class OauthBearerFilter
+        implements SaslAuthenticateRequestFilter {
+
+    private final SaslServer saslServer;
+
+    public OauthBearerFilter(SaslServer saslServer) {
+        this.saslServer = saslServer;
+    }
+
+    @Override
+    public CompletionStage<RequestFilterResult> onSaslAuthenticateRequest(short apiVersion, RequestHeaderData header,
+                                                                          SaslAuthenticateRequestData request, FilterContext context) {
+        try {
+            saslServer.evaluateResponse(request.authBytes());
+        }
+        catch (Exception e) {
+            SaslAuthenticateResponseData failedResponse = new SaslAuthenticateResponseData()
```

Comment:
Consider logging the exception stack trace, maybe at debug for problem diagnosis purposes.